org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -406,7 +406,7 @@ public CGenerator(FileConfig fileConfig, ErrorReporter errorReporter) {
     public void accommodatePhysicalActionsIfPresent() {
         // If there are any physical actions, ensure the threaded engine is used and that
         // keepalive is set to true, unless the user has explicitly set it to false.
-        List<Action> physicalActions = ASTUtils.allPhysicalActions(this.reactors);
+        List<Action> physicalActions = this.getPhysicalActions();
         if (!physicalActions.isEmpty()) {
```

Comment:
Here the list is built also only to fetch at most the first element... Consider using stream methods, eg
```java
Optional<Action> firstPhysicalAction = reactors.stream()
                 .flatMap(r -> allActions(r).stream())
                 .filter(act-> act.getOrigin() == ActionOrigin.PHYSICAL)
                 .findFirst()
```
These can often replace loops and read much less complex.

Reply:
Alternatively, I can just revert this change or make `getPhysicalActions` return an iterator - unsure if that would be satisfactory under Java